proof/refine/ARM/Ipc_R.thy

```diff
@@ -264,11 +264,7 @@ lemmas unifyFailure_discard2
 lemma deriveCap_not_null:
   "\<lbrace>\<top>\<rbrace> deriveCap slot cap \<lbrace>\<lambda>rv. K (rv \<noteq> NullCap \<longrightarrow> cap \<noteq> NullCap)\<rbrace>,-"
   apply (simp add: deriveCap_def split del: if_split)
-  apply (case_tac cap)
-          apply (simp_all add: Let_def isCap_simps)
-  apply wp
-  apply simp
-  done
+  by (case_tac cap; wpsimp simp: isCap_simps)
```

Comment:
This is an example of the fourth category of proof changes from the PR description; many of the cases here were previously solved by `simp`, now they are all solved by `wp`.